clients/python/girder_client/__init__.py

```diff
@@ -1012,6 +1012,42 @@ def downloadFolderRecursive(self, folderId, dest, sync=False):
             if len(items) < DEFAULT_PAGE_LIMIT:
                 break
 
+    def downloadResource(self, resourceId, dest, resourceType='folder', sync=False):
+        """
+        Download a folder recursively from Girder into a local directory.
+
+        :param resourceId: ID or path of the resource to download.
```

Comment:
I'd like this to be more clear.  I looked through the overall file, and there is a mix of `resourceId` type params where some are an Id only and some are an Id or a path.  It would be nice to unify these, which is probably out of scope here.

Even better might be to rename these type of params to be `resourceIdentifier` (avoiding `resourceId` as this can be an Id or a path) and have the docstring indicate that they can be an ID or a path--this would be a breaking change, but it is small enough that we could get it into the 2.0 branch reasonably quickly.

There's a good docstring in `resourceLookup` that can be the basis here, though that one doesn't include `folder`, probably because you can't look up a folder by name?
 
        :param path: The path of the resource. The path must be an absolute
            Unix path starting with either "/user/[user name]" or
            "/collection/[collection name]".

My ideal (breaking change for other methods, but OK for this one) would be

    : param resourceIdentifier: resource Id or Girder resource path.  The path must be an absolute
            Unix path starting with either "/user/[user name]" or "/collection/[collection name]".

This is fine as a simple fix and a new issue.  I'd be happy to do the cleanup.